### PR TITLE
Get latest release without version number and use correct beta download link

### PR DESCRIPTION
**Pull request by naturecodevoid**
This PR will not work for normal IPA or beta IPA since GitHub's API does not allow a pre-release to be the latest release, and because a beta build has not been triggered. However, you can test it by changing the repos to `naturecodevoid/SideStore`.

### CODE REVIEW

**Review by JoeMatt — APPROVED** [review]

**Comment by upal212** [comment]
Link should be : https://github.com/SideStore/SideStore/releases/download/nightly/SideStore.ipa

**Comment by naturecodevoid** [comment]
Yep, but nightly releases probably won’t be stable enough to have on the website. There will be a way to switch release channels in app (post release) using SideSource or something

**Review by SternXD — APPROVED** [review]

**Comment by f1shy-dev** [comment]
Where did the new parcel package come from?

**Comment by naturecodevoid** [comment]
I had to install it to run parcel, not sure why.

**Comment by f1shy-dev** [comment]
Converting to draft as @upal212's previous PR/commit had reverted changes I made on the buttons.

I will need to take a look at this to use the proper buttons (basically I abstracted the buttons into a class, and I also removed the <div> on buttons, and this fix being removed means that you have to click the text on a button for it to work rather than the actual button)

File: src/index.js [hunk 1]
```diff
@@ -96,6 +96,18 @@ window.apps = apps;
 
   console.log(`Loaded apps marquee with ${apps.length} items.`);
 
+  const latestIPADownload = document.querySelector('#latest-ipa-download');
+  const latestRelease = await (await fetch('https://api.github.com/repos/SideStore/SideStore/releases/latest')).json();
+  if ('assets' in latestRelease)
+    for (const asset of latestRelease.assets) {
+      if (asset.name.endsWith('.ipa')) {
+        latestIPADownload.href = asset.browser_download_url;
+        console.log('Found latest release and .ipa', asset, latestRelease);
```
Comment (f1shy-dev) [inline src/index.js]: I will also implement some loading indicators on this, for low-bandwidth connections.

File: src/index.html [hunk 2]
```diff
@@ -172,7 +172,7 @@ <h1 class="mb-2 font-['Poppins'] text-xl font-medium text-white">SideStore IPA</
             class="flex items-center justify-center rounded-full bg-purple-500 px-7 py-2.5 transition-colors duration-150 ease-in-out hover:bg-purple-600"
           >
             <a
-              href="https://nightly.link/SideStore/SideStore/workflows/build/develop/SideStore.ipa.zip"
+              href="https://github.com/SideStore/SideStore/releases/download/beta/SideStore.ipa"
```
Comment (f1shy-dev) [inline src/index.html:175]: We need to add a similar warning as to the one before about the beta/nightly IPAs.

Comment (f1shy-dev) [reply src/index.html:175]: Also todo here, change the buttons to the new format.

File: src/index.js (hunk 1 above)
Comment (naturecodevoid) [reply src/index.js]: These changes are not needed and outdated, since we can just use a direct latest download link.

File: src/index.html (hunk 2 above)
Comment (naturecodevoid) [reply src/index.html:175]: I think we should only have beta on the site, and keep nightly for advanced users/testers. There will be a way to switch in-app or something anyways.

Comment (f1shy-dev) [reply src/index.html:175]: Is beta part of the build system?

Comment (naturecodevoid) [reply src/index.html:175]: Yes, but it hasn't been triggered. There is also an action for stable builds

**Pull request closed without merging**